### PR TITLE
Add ebml.xml file

### PR DESCRIPTION
**Pull request by StefH**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ bootstrap.mak
 /runtimes.mak
 metadata.min.js
 /rfc8794.xml
+/.vs
```
Comment (robUx4) [inline .gitignore:8]: This is not referenced anywhere in our code. (you can add it to your local gitignore instead)

File: ebml.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		<documentation lang="en" purpose="definition">A string that describes the type of document that follows this EBML header. 'matroska' in our case or 'webm' for webm files.</documentation>
+	</element>
+	<element name="DocTypeVersion" path="\EBML\DocTypeVersion" id="0x4287" type="uinteger" range="1-8" default="1" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">The version of DocType interpreter used to create the file.</documentation>
+	</element>
+	<element name="DocTypeReadVersion" path="\EBML\DocTypeReadVersion" id="0x4285" type="uinteger" range="1-8" default="1" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">The minimum DocType version an interpreter has to support to read this file.</documentation>
+	</element>
+
+	<element name="SignatureSlot" path="\EBML\SignatureSlot" id="0x1B538667" type="master">
```
Comment (robUx4) [inline ebml.xml:27]: Same remark as before these signature elements don't belong here.

File: ebml.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		</documentation>
+	</element>
+	<element name="SignedElement" path="\EBML\SignatureElements\SignatureElementList\SignedElement" id="0x6532" type="binary" >
+		<documentation lang="en" purpose="definition">An element ID whose data will be used to compute the signature.</documentation>
+	</element>
+
+	<element name="Void" path="\(-\)Void" id="0xEC" type="binary">
+		<documentation lang="en" purpose="definition">Used to void damaged data, to avoid unexpected behaviors when using damaged data. The content is discarded. Also used to reserve space in a sub-element for later use.</documentation>
+	</element>
+	<element name="CRC-32" path="\(1-\)CRC-32" id="0xBF" type="binary">
```
Comment (robUx4) [inline ebml.xml:58]: Missing the `maxOccurs="1"`

File: ebml.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+<EBMLSchema xmlns="urn:ietf:rfc:8794" docType="ebml" version="1">
+	<element name="EBML" path="\EBML" id="0x1A45DFA3" type="master" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">Set the EBML characteristics of the data to follow. Each EBML document has to start with this.</documentation>
+	</element>
+	<element name="EBMLVersion" path="\EBML\EBMLVersion" id="0x4286" type="uinteger" range="1-8" default="1" minOccurs="1" maxOccurs="1">
```
Comment (robUx4) [inline ebml.xml:5]: The range is `not 0`.

File: ebml.xml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		<documentation lang="en" purpose="definition">The version of EBML parser used to create the file.</documentation>
+	</element>
+	<element name="EBMLReadVersion" path="\EBML\EBMLReadVersion" id="0x42F7" type="uinteger" range="1" default="1" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">The minimum EBML version a parser has to support to read this file.</documentation>
+	</element>
+	<element name="EBMLMaxIDLength" path="\EBML\EBMLMaxIDLength" id="0x42F2" type="uinteger" range="4" default="4" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">The maximum length of the IDs you'll find in this file (4 or less in Matroska).</documentation>
+	</element>
+	<element name="EBMLMaxSizeLength" path="\EBML\EBMLMaxSizeLength" id="0x42F3" type="uinteger" range="1-8" default="8" minOccurs="1" maxOccurs="1">
```
Comment (robUx4) [inline ebml.xml:14]: The range is `not 0`.

File: ebml.xml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		<documentation lang="en" purpose="definition">The maximum length of the sizes you'll find in this file (8 or less in Matroska). This does not override the element size indicated at the beginning of an element. Elements that have an indicated size which is larger than what is allowed by EBMLMaxSizeLength shall be considered invalid.</documentation>
+	</element>
+	<element name="DocType" path="\EBML\DocType" id="0x4282" type="string" default="ebml" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">A string that describes the type of document that follows this EBML header. 'matroska' in our case or 'webm' for webm files.</documentation>
+	</element>
+	<element name="DocTypeVersion" path="\EBML\DocTypeVersion" id="0x4287" type="uinteger" range="1-8" default="1" minOccurs="1" maxOccurs="1">
```
Comment (robUx4) [inline ebml.xml:20]: The range is `not 0`.

File: ebml.xml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+		<documentation lang="en" purpose="definition">The version of DocType interpreter used to create the file.</documentation>
+	</element>
+	<element name="DocTypeReadVersion" path="\EBML\DocTypeReadVersion" id="0x4285" type="uinteger" range="1-8" default="1" minOccurs="1" maxOccurs="1">
```
Comment (robUx4) [inline ebml.xml:23]: The range is `not 0`.

File: ebml.xml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+		<documentation lang="en" purpose="definition">The minimum EBML version a parser has to support to read this file.</documentation>
+	</element>
+	<element name="EBMLMaxIDLength" path="\EBML\EBMLMaxIDLength" id="0x42F2" type="uinteger" range="4" default="4" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">The maximum length of the IDs you'll find in this file (4 or less in Matroska).</documentation>
+	</element>
+	<element name="EBMLMaxSizeLength" path="\EBML\EBMLMaxSizeLength" id="0x42F3" type="uinteger" range="1-8" default="8" minOccurs="1" maxOccurs="1">
+		<documentation lang="en" purpose="definition">The maximum length of the sizes you'll find in this file (8 or less in Matroska). This does not override the element size indicated at the beginning of an element. Elements that have an indicated size which is larger than what is allowed by EBMLMaxSizeLength shall be considered invalid.</documentation>
+	</element>
+	<element name="DocType" path="\EBML\DocType" id="0x4282" type="string" default="ebml" minOccurs="1" maxOccurs="1">
```
Comment (robUx4) [inline ebml.xml:17]: missing `length=">0"` but in a proper XML way, so `length="&gt;0"`

File: ebml.xml [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	</element>
+	<element name="SignatureElementList" path="\EBML\SignatureElements\SignatureElementList" id="0x7E7B" type="master">
+		<documentation lang="en" purpose="definition">
+			A list consists of a number of consecutive elements that represent one case where data is used in signature. Ex: Cluster|Block|BlockAdditional means that the BlockAdditional of all Blocks in all Clusters is used for encryption.
+		</documentation>
+	</element>
+	<element name="SignedElement" path="\EBML\SignatureElements\SignatureElementList\SignedElement" id="0x6532" type="binary" >
+		<documentation lang="en" purpose="definition">An element ID whose data will be used to compute the signature.</documentation>
+	</element>
+
```
Comment (robUx4) [inline ebml.xml:54]: It's missing `DocTypeExtension`, `DocTypeExtensionName`, `DocTypeExtensionVersion`.

File: ebml.xml (hunk 3 above)
Comment (robUx4) [reply ebml.xml:58]: Missing `length="4"`

File: ebml.xml [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+	<element name="Void" path="\(-\)Void" id="0xEC" type="binary">
+		<documentation lang="en" purpose="definition">Used to void damaged data, to avoid unexpected behaviors when using damaged data. The content is discarded. Also used to reserve space in a sub-element for later use.</documentation>
+	</element>
+	<element name="CRC-32" path="\(1-\)CRC-32" id="0xBF" type="binary">
+		<documentation lang="en" purpose="definition">The CRC is computed on all the data of the Master element it's in. The CRC element should be the first in it's parent master for easier reading. All level 1 elements should include a CRC-32. The CRC in use is the IEEE CRC32 Little Endian.</documentation>
+	</element>
+</EBMLSchema>
```
Comment (robUx4) [inline ebml.xml:61]: Git doesn't like files that don't end up with an empty line. So you should add one to make it look nicer.

**Review by robUx4 — CHANGES_REQUESTED** [review]
Thanks a lot for taking the time to do this. There are some differences with the values from the RFC and some elements missing.

**Comment by robUx4** [comment]
Fixes #391

File: .gitignore (hunk 1 above)
Comment (StefH) [reply .gitignore:8]: fixed

File: ebml.xml (hunk 3 above)
Comment (StefH) [reply ebml.xml:58]: ok

File: ebml.xml (hunk 2 above)
Comment (StefH) [reply ebml.xml:27]: ok

File: ebml.xml (hunk 5 above)
Comment (StefH) [reply ebml.xml:14]: ok

File: ebml.xml (hunk 6 above)
Comment (StefH) [reply ebml.xml:20]: ok

File: ebml.xml (hunk 7 above)
Comment (StefH) [reply ebml.xml:23]: ok

File: ebml.xml (hunk 8 above)
Comment (StefH) [reply ebml.xml:17]: ok

File: ebml.xml (hunk 10 above)
Comment (StefH) [reply ebml.xml:61]: ok

**Comment by StefH** [comment]
I think that all comments are resolved?

**Comment by robUx4** [comment]
There's still the range you left a `1-8`. After that it's good to merge.
Thanks a lot.

File: ebml.xml (hunk 5 above)
Comment (StefH) [reply ebml.xml]: Fixed

**Comment by robUx4** [comment]
Excellent ! Thanks a lot for doing this, and so promptly !

**Comment by StefH** [comment]
@robUx4 
I did also fix the range from EBMLMaxIDLength  to >=4

**Comment by StefH** [comment]
@robUx4 
You were too fast ;-)

What about  `EBMLMaxIDLength` ?
That should be 
``` xml
range="&ge;4"
```

?

**Comment by robUx4** [comment]
The merged version of `EBMLMaxIDLength` has `range="&gt;4" default="4"` which matches the RFC text.

**Comment by StefH** [comment]
The specification.markdown has different value?
https://github.com/ietf-wg-cellar/ebml-specification/blob/master/specification.markdown#ebmlmaxidlength-element

**Comment by robUx4** [comment]
It could be interesting to adapt the code that transforms the Matroska XML file to markdown for the RFC, to this XML file. The result should match almost exactly the markdown used to generate the RFC.

**Comment by robUx4** [comment]
> 
> 
> The specification.markdown has different value?
> https://github.com/ietf-wg-cellar/ebml-specification/blob/master/specification.markdown#ebmlmaxidlength-element

Ah `>` vs `>=` ? I'll fix it.

**Comment by StefH** [comment]
OK, then you can abandon my PR.

**Comment by StefH** [comment]
BTW : I'm building a C# based Matroska parser, and I'm using these xml files to generate the specifications and classes using T4 templating.
https://github.com/StefH/Matroska/blob/ebml/src/Matroska/MatroskaSpecification.tt

**Comment by robUx4** [comment]
Very nice. That's definitely why it's good to have machine readable specs.
In my case I use it to generate libmatroska and libmatroska2 code and it makes refactoring easy and safe.

**Pull request merged**